src/meshlink.c

```diff
@@ -2387,66 +2077,40 @@ char *meshlink_invite_ex(meshlink_handle_t *mesh, meshlink_submesh_t *submesh, c
 
 	free(fingerprint);
 
-	// Create a file containing the details of the invitation.
-	snprintf(filename, sizeof(filename), "%s" SLASH "invitations" SLASH "%s", mesh->confbase, cookiehash);
-	int ifd = open(filename, O_RDWR | O_CREAT | O_EXCL, 0600);
+	/* Construct the invitation file */
+	uint8_t outbuf[4096];
+	packmsg_output_t inv = {outbuf, sizeof(outbuf)};
 
-	if(!ifd) {
-		logger(mesh, MESHLINK_DEBUG, "Could not create invitation file %s: %s\n", filename, strerror(errno));
-		meshlink_errno = MESHLINK_ESTORAGE;
-		pthread_mutex_unlock(&(mesh->mesh_mutex));
-		return NULL;
-	}
+	packmsg_add_uint32(&inv, MESHLINK_INVITATION_VERSION);
+	packmsg_add_str(&inv, name);
+	packmsg_add_str(&inv, s ? s->name : CORE_MESH);
+	packmsg_add_int32(&inv, DEV_CLASS_UNKNOWN); /* TODO: allow this to be set by inviter? */
 
-	FILE *f = fdopen(ifd, "w");
+	/* TODO: Add several host config files to bootstrap connections */
+	config_t configs[5] = {NULL};
+	int count = 0;
 
-	if(!f) {
-		abort();
+	if(config_read(mesh, mesh->self->name, &configs[count])) {
+		count++;
 	}
 
-	// Fill in the details.
-	fprintf(f, "Name = %s\n", name);
+	/* Append host config files to the invitation file */
+	packmsg_add_array(&inv, count);
 
-	if(s) {
-		fprintf(f, "SubMesh = %s\n", s->name);
+	for(int i = 0; i < count; i++) {
```

Comment:
I think adding host config files other than inviter node to the invitation file looks like a TODO item, however I think we should write a condition where if the invitee node belongs to a submesh we should only append host node config files of that submesh only to the invitation.